### PR TITLE
Jsc/my customer intro

### PR DESCRIPTION
**Pull request by ickhov**

### CODE REVIEW

File: CCIA/Controllers/MyCustomersController.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            myCustomer.Name = model.MyCustomer.Name;
+            myCustomer.Address1 = model.MyCustomer.Address1;
+            myCustomer.Address2 = model.MyCustomer.Address2;
+            myCustomer.City = model.MyCustomer.City;
+            myCustomer.CountryId = model.MyCustomer.CountryId;
+
+            // 10 is Canada
+            // 58 is United States
+            if (myCustomer.CountryId == 10 || myCustomer.CountryId == 58) {
+                myCustomer.CountyId = model.MyCustomer.CountyId;
```
Comment (jcubbage) [inline CCIA/Controllers/MyCustomersController.cs]: Country of 10 (Canada) gets you a state/province, but not a valid county. Either move county into a new If w/just 58, or separate these into 2 if without the "or".

File: CCIA/Controllers/MyCustomersController.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            myCustomerToEdit.Name = model.MyCustomer.Name;
+            myCustomerToEdit.Address1 = model.MyCustomer.Address1;
+            myCustomerToEdit.Address2 = model.MyCustomer.Address2;
+            myCustomerToEdit.City = model.MyCustomer.City;
+            
+            myCustomerToEdit.CountryId = model.MyCustomer.CountryId;
+
+            // 10 is Canada
+            // 58 is United States
+            if (myCustomerToEdit.CountryId == 10 || myCustomerToEdit.CountryId == 58) {
```
Comment (jcubbage) [inline CCIA/Controllers/MyCustomersController.cs]: Same note about county goes here. In reality, county should only be completed if Country = 58 (US) and State = 102 (California).

File: CCIA/Models/MyCustomerViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            
+            var model = await _dbContext.MyCustomers.Where(a => a.Id == id)
+                .Include(e => e.State)
+                .Include(e => e.County)
+                .Include(e => e.Country)
+                .Include(e => e.Organization)
+                .FirstOrDefaultAsync();
+
+            var stateProvinces = await _dbContext.StateProvince
+                .OrderBy(s => s.StateProvinceName)
```
Comment (jcubbage) [inline CCIA/Models/MyCustomerViewModel.cs]: If you sort by country first, the drop down list will separate out nicely.

File: CCIA/Views/MyCustomers/Create.cshtml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    <script type="application/javascript">
+
+        $(document).ready(function () {
+            $('#MyCustomer_CountryId').change(function () {               
+                var country = $("#MyCustomer_CountryId option:selected");
+                showStates(country.text());
+            });
+        });
+
+        function showStates(country)
```
Comment (jcubbage) [inline CCIA/Views/MyCustomers/Create.cshtml]: This is fine for now. But we should create an issue to revisit it. I made a change in the SeedsCrud branch that makes this work better.

File: CCIA/Controllers/MyCustomersController.cs (hunk 1 above)
Comment (ickhov) [reply CCIA/Controllers/MyCustomersController.cs]: I separated the conditions into 2 separate statements.

`// if Canada or United states is selected, then state should be filled
            if (myCustomerToEdit.CountryId == 10 || myCustomerToEdit.CountryId == 58) {
                myCustomerToEdit.StateId = model.MyCustomer.StateId;
            } else {
                myCustomerToEdit.StateId = 0;
            }

            // if United States and California are selected, then county should be filled
            if (myCustomerToEdit.CountryId == 58 && myCustomerToEdit.StateId == 102) {
                myCustomerToEdit.CountyId = model.MyCustomer.CountyId;
            } else {
                 myCustomerToEdit.CountyId = 0;
            }`

File: CCIA/Controllers/MyCustomersController.cs (hunk 2 above)
Comment (ickhov) [reply CCIA/Controllers/MyCustomersController.cs]: Same here.

File: CCIA/Models/MyCustomerViewModel.cs (hunk 3 above)
Comment (ickhov) [reply CCIA/Models/MyCustomerViewModel.cs]: I sorted by state names first and by country id in descending order so US states can be on top.

File: CCIA/Views/MyCustomers/Create.cshtml (hunk 4 above)
Comment (ickhov) [reply CCIA/Views/MyCustomers/Create.cshtml]: I'll wait for you to push the new SeedsCrud to do this.

File: CCIA/Models/MyCustomerViewModel.cs (hunk 3 above)
Comment (jcubbage) [reply CCIA/Models/MyCustomerViewModel.cs]: Close. Take a look at the New Customer page. Canada and States are mixed. Also, when you want multiple sort orders, you need .OrderBy() followed by .ThenBy().
This is from new customer:

<option value="105">Arkansas</option>
--
  | <option value="161">British Columbia, Canada</option>
  | <option selected="selected" value="102">California</option>
  | <option value="107">Colorado</option>

Comment (ickhov) [reply CCIA/Models/MyCustomerViewModel.cs]: Ok this new commit should work now.

**Pull request merged**